deeppavlov/utils/socket/socket.py

```diff
@@ -0,0 +1,175 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import json
+import os
+import socket
+from logging import getLogger
+from pathlib import Path
+from typing import Dict, Optional, Tuple, Union
+
+from deeppavlov.core.agent.dialog_logger import DialogLogger
+from deeppavlov.core.commands.infer import build_model
+from deeppavlov.core.commands.utils import parse_config
+from deeppavlov.core.common.chainer import Chainer
+from deeppavlov.core.common.file import read_json
+from deeppavlov.core.common.paths import get_settings_path
+from deeppavlov.core.data.utils import check_nested_dict_keys, jsonify_data
+
+SOCKET_CONFIG_FILENAME = 'socket_config.json'
+
+
+def get_socket_params(socket_config_path: Path, model_config: Path) -> Dict:
+    socket_config = read_json(socket_config_path)
+    model_config = parse_config(model_config)
+
+    socket_params = socket_config['common_defaults']
+
+    if check_nested_dict_keys(model_config, ['metadata', 'labels', 'server_utils']):
+        model_tag = model_config['metadata']['labels']['server_utils']
+        if model_tag in socket_config['model_defaults']:
+            model_defaults = socket_config['model_defaults'][model_tag]
+            socket_params.update(model_defaults)
+    return socket_params
+
+
+class SocketServer:
+    """
+    Class with socket server. The data received by the socket is processed in the deeppavlov model. Sends model
+    response back as dictionary with two keys - 'status' and 'payload':
+    response['status']: str - 'OK' if data processed successfully, else - error message.
+    response['payload']: str - model response dumped to bytes. Empty if an error occured.
+    """
+    _bind_address: Union[Tuple, str]
+    _launch_msg: str
+    _loop: asyncio.AbstractEventLoop
+    _model: Chainer
+    _params: Dict
+    _socket: socket.socket
+    _socket_type: str
+
+    def __init__(self, model_config: Path, socket_type: str, port: Optional[int] = None,
+                 socket_file: Optional[str] = None):
+        socket_config_path = get_settings_path() / SOCKET_CONFIG_FILENAME
+        self._params = get_socket_params(socket_config_path, model_config)
+        self._socket_type = socket_type or self._params['socket_type']
+
+        if self._socket_type == 'TCP':
+            host = self._params['host']
+            port = port or self._params['port']
+            self._address_family = socket.AF_INET
+            self._launch_msg = f'{self._params["binding_message"]} http://{host}:{port}'
+            self._bind_address = (host, port)
+        elif self._socket_type == 'UNIX':
+            self._address_family = socket.AF_UNIX
+            self._bind_address = socket_file or self._params['unix_socket_file']
+            if os.path.exists(self._bind_address):
+                os.remove(self._bind_address)
```

Comment:
Can `self._bind_address` be a `Path` here? No need to use `os.path` and `os.remove` this way